syntax = "proto3";

package analysis;

service Analysis 
{
  // Sends a greeting
  rpc UploadScript (UploadScriptRequest) returns (UploadScriptReply) {}
}

message UploadScriptRequest
{
    string script_name = 1;
    string script = 2;
}

message UploadScriptReply
{
    bool upload_success = 1;
    oneof response_oneof
    {
        string script_uuid = 2;
        string fail_reason = 3;
    }
}